lib/components/prometheusmetrics.js

```diff
@@ -208,6 +208,122 @@ PrometheusMetrics.prototype._register_process_metrics = function() {
     this.refresh();
 };
 
+/**
+ * Registers some exported metrics that relate to operations of the embedded
+ * matrix-js-sdk. In particular, a metric is added that counts the number of
+ * calls to client API endpoints made by the client library.
+ */
+PrometheusMetrics.prototype.registerMatrixSdkMetrics = function() {
+    var callCounts = this.addCounter({
+        name: "matrix_api_calls",
+        help: "Count of the number of Matrix client API calls made",
+    });
+
+    /*
+     * We'll now annotate a bunch of the methods in MatrixClient to keep counts
+     * of every time they're called. This seems to be neater than trying to
+     * intercept all HTTP requests and try to intuit what internal method was
+     * invoked based on the HTTP URL.
+     */
+
+    var matrixClientPrototype = require("matrix-js-sdk").MatrixClient.prototype;
+
+    var CLIENT_METHODS = [
+        "ban",
+        "createAlias",
```

Comment:
That was my original plan, but then I hit a snag. The snag is in summary - REST.

The URLs visible at the http `request` function contain embedded in them lots of IDs - room IDs, user IDs, room state event types, etc... In order to unpick those and decode what the actual API endpoint is would require keeping a big list of all the various API endpoints with glob patterns or similar to unpick the IDs out of the URLs and then turn them into method names. E.g. dendron keeps such a list; this entire file is dedicated to such configuration:

https://github.com/matrix-org/dendron/blob/develop/src/github.com/matrix-org/dendron/proxy/paths.go

I decided that maintaining another list like that would be *even harder* than just keeping a list of SDK method names, because at least now I don't have to parse URLs or adapt to shape changes there, only names of methods within the MatrixClient SDK class itself.